src/server/mcp.ts

```diff
@@ -609,15 +598,15 @@ export class McpServer {
       description = rest.shift() as string;
     }
 
-    let argsSchema: PromptArgsRawShape | undefined;
+    let argsSchema: StandardSchemaV1 | undefined;
     if (rest.length > 1) {
-      argsSchema = rest.shift() as PromptArgsRawShape;
+      argsSchema = rest.shift() as StandardSchemaV1 | undefined;
     }
 
-    const cb = rest[0] as PromptCallback<PromptArgsRawShape | undefined>;
+    const cb = rest[0] as NullaryPromptCallback | UnaryPromptCallback
     this._registeredPrompts[name] = {
       description,
-      argsSchema: argsSchema === undefined ? undefined : z.object(argsSchema),
+      argsSchema,
```

Comment:
maybe this could be non-breaking by doing

```suggestion
      argsSchema: argsSchema === undefined
        ? undefined
        : argsSchema?.['~standard']
          ? argsSchema
          : z.object(argsSchema),
```

And resurrecting `PromptArgsRawShape` as ts union type input